File: day15/DayFifteen.java
public class DayFifteen
{
	public static void main(String args[])
	{
		int arr[] = {1,5,-7,-8,10,11,9,-20,11};
		DayFifteen maxSubArraySum = new DayFifteen();
		System.out.println(maxSubArraySum.kadane(arr));
	}

	public int kadane(int arr[])
	{
		int max_sum_curr = 0;
		int max_sum_till_now = 0;

		for(int i=0;i<arr.length;i++)
		{
			max_sum_curr = max_sum_curr + arr[i];
			if(max_sum_curr < 0)
			{
				max_sum_curr = 0;
			}
			else if(max_sum_till_now < max_sum_curr)
			{
				max_sum_till_now = max_sum_curr;
			}
		}

		return max_sum_till_now;
	}
}

File: day15/day_fifteen.cpp
# include <iostream>

using namespace std;

int kadane(int arr[],int n)
{
	int max_sum_till_here = 0;
	int max_sum_now = 0;
	for(int i=0;i<n;i++)
	{
		max_sum_now = max_sum_now + arr[i];

		if(max_sum_now < 0)
		{
			max_sum_now = 0;
		}
		else if(max_sum_till_here < max_sum_now)
		{
			max_sum_till_here = max_sum_now;
		}
	}

	return max_sum_till_here;
}

int main()
{
	int arr[] = {1,5,-7,-8,10,11,9,-20,11};
	int n = sizeof(arr)/sizeof(arr[0]);
	cout << kadane(arr,n) << endl;
	return 0;
}